src/extension.ts

```diff
@@ -199,6 +200,23 @@ export async function activate(context: vscode.ExtensionContext, perfStats: { lo
     ext.managedIdentityBranchDataProvider = new ManagedIdentityBranchDataProvider();
     ext.v2.api.resources.registerAzureResourceBranchDataProvider(AzExtResourceType.ManagedIdentityUserAssignedIdentities, ext.managedIdentityBranchDataProvider);
 
+
+
+    // Register the exportAuthRecord callback for session changes
+    registerEvent(
+        'treeView.onDidChangeSessions',
+        vscode.authentication.onDidChangeSessions,
+        exportAuthRecord
+    );
+
+    // Also call exportAuthRecord once on activation to ensure the auth record is exported at least once
+    // (onDidChangeSessions is not guaranteed to fire on activation)
+    void exportAuthRecord({
```

Comment:
Should we await for this function execution?

Reply:
This shouldn't be awaited, as it would slow the execution of the extension activation which is an extremely critical path.

Reply:
if its guaranteed to execute async here, then don't see any need to await here.

Reply:
not awaiting here.